src/diffusers/models/attention.py

```diff
@@ -151,8 +155,10 @@ def __init__(
             self.norm1 = AdaLayerNorm(dim, num_embeds_ada_norm)
         elif self.use_ada_layer_norm_zero:
             self.norm1 = AdaLayerNormZero(dim, num_embeds_ada_norm)
-        else:
+        elif caption_channels is None:
```

Comment:
`caption_channels` has nothing to do with `norm_elementwise_affine` no? 

Can'n we just provide a `norm_eps` and `elementwise_affine` parameter to the `__init__`?